File: app/javascript/channels/search_counts_channel.js
/* import consumer from "./consumer";
// ...

const chatChannel = consumer.subscriptions.create("SearchCountsChannel", {
  connected() {
    console.log("Connected to ChatChannel");
  },
  received(data) {
    console.log("Received data:", data);
  }
}); */

// Manually subscribe to the channel
// chatChannel.perform("subscribed");

// Broadcast a message from Rails console
// ActionCable.server.broadcast("chat", message: "Hello, world!");

File: app/channels/search_counts_channel.rb
class SearchCountsChannel < ApplicationCable::Channel
  def subscribed
    stream_from 'search_counts_channel'
  end

  def update_search_counts(data)
    user_id = data['user_id']
    matching_count = data['matching_count']
    non_matching_count = data['non_matching_count']

    # Update counts based on data received

    # Broadcast the updated counts to all subscribed clients
    ActionCable.server.broadcast('search_counts_channel', {
      user_id: user_id,
      matching_count: matching_count,
      non_matching_count: non_matching_count
    })
  end
end
